libs/ledger-live-common/src/families/solana/js-preload.test.ts

```diff
@@ -0,0 +1,14 @@
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { hydrate } from "./js-preload";
+import { SolanaPreloadData } from "./types";
+
+describe("hydrate", () => {
+  it.each([undefined, null, {}, { version: null }, { version: undefined }, {}])(
+    "should return undefined if data is corrupted (%p value)",
+    value => {
+      expect(hydrate(value as unknown as SolanaPreloadData, {} as CryptoCurrency)).toEqual(
```

Comment:
Why casting it as `unknown`? You should be able to call "hydrate" with the accepted parameter type (i.e. `null` should never be able to call this function, as well as a non string `version` value)

Reply:
Please check description and Sentry, this value is not intended and should not happen. It can't be reproduced manually

Reply:
Do you think I should add more cases ?

Reply:
Sure, can do ! :) 
I think it's supposed to be the base behavior of hydrate and it should have been this way since the beginning